updateParamGroups.iLogicVb

```diff
@@ -1,35 +1,40 @@
 ﻿Option Explicit On
 Imports Inventor.ObjectTypeEnum
 Imports System.Text.RegularExpressions
+Public Sub Main()
+	
+	Dim oUpdateParamGroup As updateParamGroup = New updateParamGroup()	
+End Sub
 
-Private Class updateParamGroup
-	Private doc As Document
+Public Class updateParamGroup
+	
 	Private oParams As Parameters
 	Private paramGroups As CustomParameterGroups
-
 	Private groupNameDelimiter As String = ":"
 	Private initialName As String
 
-	Private Sub Main()
+	Public Sub New()
 		addEventHandler()
 	End Sub
 
-	Private Sub oModelEvent_ParameterChange(
-		DocumentObject As _Document,
+	Private Sub oModelingEvents_ParameterChange(
+		oDoc As Inventor.Document,
 		Parameter As Parameter,
 		BeforeOrAfter As EventTimingEnum,
 		Context As NameValueMap,
 		ByRef HandlingCode As HandlingCodeEnum
 	)
+	
 		If BeforeOrAfter = kBefore Then
-			initialName = Parameter.Name
-			getDocumentMembers()
-			Return
+			Exit Sub
+			'MsgBox("The event has fired, BeforeOrAfter = kBefore")
 		Else If Not BeforeOrAfter = kAfter Then
-			Return
+			'MsgBox("The event has fired, BeforeOrAfter = kAfter")
+			initialName = Parameter.Name
+			getDocumentMembers(oDoc)
 		End If
```

Comment:
I wrote this in a way that is a little confusing. I don't think there's actually 'response' for anything other than `kBefore` or `kAfter`, and in any case `initialName` needs to be set under `kBefore`. This `Else If Not` was just to protect the script against my ignorance lol, and a habit of early returning that was instilled into me, but can probably be removed altogether, provided what's in the code block be restored into the `kBefore`